pkg/assessor/manifest/manifest.go

```diff
@@ -55,7 +56,20 @@ func AddAcceptanceKeys(keys []string) {
 	}
 }
 
+func setSensitivePatterns() error {
+	pat := fmt.Sprintf(`.*(?i)%s.*`, strings.Join(suspiciousEnvKey, "|"))
+	r, err := regexp.Compile(pat)
+	if err != nil {
+		return fmt.Errorf("compile suspicious key: %w", err)
+	}
+	suspiciousCompiler = r
```

Comment:
`regexp.Compile` is very slow, so I want to avoid compile it each time.
https://github.com/golang/go/issues/26623#issuecomment-408518236

Reply:
Should it be compiled before hand in `GetAssessments` and passed every time `results, err := assessor.Assess(files)` is called? or what's your idea @tomoyamachi ?

Reply:
tests for manifest checking sensitive vars fail with this change;

```
=== RUN   TestSensitiveVars
--- FAIL: TestSensitiveVars (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x5093d0]

goroutine 6 [running]:
testing.tRunner.func1.2({0x5e2fc0, 0x789c30})
        /usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x5e2fc0, 0x789c30})
        /usr/local/go/src/runtime/panic.go:838 +0x207
regexp.(*Regexp).doExecute(0x0?, {0x0?, 0x0?}, {0x0?, 0xffffffffffffffff?, 0x0?}, {0xc000166db0?, 0xc0000506b0?}, 0x5e20c0?, 0x0, ...)
        /usr/local/go/src/regexp/exec.go:527 +0x90
regexp.(*Regexp).doMatch(...)
        /usr/local/go/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
        /usr/local/go/src/regexp/regexp.go:531
github.com/goodwithtech/dockle/pkg/assessor/manifest.sensitiveVars({0x61f612, 0x20})
        /home/alvaro/myDockle/pkg/assessor/manifest/manifest.go:245 +0x205
github.com/goodwithtech/dockle/pkg/assessor/manifest.TestSensitiveVars(0xc000121040)
        /home/alvaro/myDockle/pkg/assessor/manifest/manifest_test.go:417 +0x2fa
testing.tRunner(0xc000121040, 0x62b890)
        /usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1486 +0x35f
FAIL    github.com/goodwithtech/dockle/pkg/assessor/manifest    0.005s

```

Reply:
Thank you for your review. I fixed it. (https://github.com/goodwithtech/dockle/pull/189/commits/1fd632bd1438d98fb03764e76b705adcc257be17)